<html>
    <script>
        // function div(a,b)
        // {
        //     if(isNaN(a) || isNaN(b))
        //     {
        //         throw new Error('A or B is NaN');
        //     }
        //     return a/b;
        // }
        // try// detect exception code
        // {
        //     let result = div(10, NaN)
        //     console.log('Result', result);
        // }
        // catch(e)//show error message and
        // //release the other code
        // {
        //     console.log(e);
        // }
        // console.log('End of Code');

        console.log('=================');
        function div1(a, b)
        {
            if(isNaN(a) || isNaN(b))
            {
                throw new Error('A or B is NaN');
            }
            return a/b;
        }
        try
        {
            let result = div1(10, NaN);
            console.log('Result', result);
        }
        catch(e)
        {
            console.log(e);
        }
        finally//finally will always excuted with exception or without exception
        {
            console.log('That is finally');
        }
        console.log('End of Code');
    </script>
</html>